Find a functionally equivalent code:com.android.internal.policy.impl.PhoneWindowManager.updateOrientationListenerLp:COMMENT
Method Modifier: internal    
Comment:/*
     * Various use cases for invoking this function
     * screen turning off, should always disable listeners if already enabled
     * screen turned on and current app has sensor based orientation, enable listeners
     * if not already enabled
     * screen turned on and current app does not have sensor orientation, disable listeners if
     * already enabled
     * screen turning on and current app has sensor based orientation, enable listeners if needed
     * screen turning on and current app has nosensor based orientation, do nothing
     */

@@ -5,13 +5,13 @@
     }
     // change of the currently visible window's orientation
     if (localLOGV)
-        Slog.v(TAG, "Screen status=" + mScreenOnEarly + ", current orientation=" + mCurrentAppOrientation + ", SensorEnabled=" + mOrientationSensorEnabled);
+        Slog.v(TAG, "mScreenOnEarly=" + mScreenOnEarly + ", mAwake=" + mAwake + ", mCurrentAppOrientation=" + mCurrentAppOrientation + ", mOrientationSensorEnabled=" + mOrientationSensorEnabled);
     boolean disable = true;
-    if (mScreenOnEarly) {
+    if (mScreenOnEarly && mAwake) {
         if (needSensorRunningLp()) {
             disable = false;
             // enable listener if not already enabled

